scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -461,6 +462,39 @@ class Router(formatOps: FormatOps) {
         verticalMultiline(leftOwner, ft)(style)
 
       // Term.Apply and friends
+      case FormatToken(T.LeftParen(), _, _)
+          if style.optIn.configStyleArguments &&
+            !style.newlines.beforeSingleArgParenLambdaParams &&
+            getLambdaAtSingleArgCallSite(leftOwner).isDefined => {
+        val lambda = getLambdaAtSingleArgCallSite(leftOwner).get
+        val (lambdaIsABlock, lambdaToken) = lambda.body match {
+          case b: Term.Block =>
+            true -> b.tokens.head
+          case _ =>
+            val arrow = lambda.tokens.find(_.is[T.RightArrow]).get
+            next(arrow).is[T.LeftBrace] -> arrow
+        }
+
+        val close = matchingParentheses(hash(formatToken.left))
+        val splitOnClosePolicy =
+          if (style.danglingParentheses.callSite && !lambdaIsABlock)
+            newlineOnTokenPolicy(close)
+          else
+            Policy(Policy.emptyPf, close.end)
+        val noSplitBeforeArrowPolicy =
+          splitOnClosePolicy.andThen(SingleLineBlock(lambdaToken))
+
+        val newlinePenalty = 3 + nestedApplies(leftOwner)
```

Comment:
This part is voodoo. I started with `1` but ended up with this after testing on `scala-repos`.

Reply:
Mostly everything in the router is vodoo 😅 it's normal to reach some magic penalty based on running the formatter on scala-repos 👍